core/src/main/scala/kafka/zk/ZkMigrationClient.scala

```diff
@@ -303,6 +304,14 @@ class ZkMigrationClient(
     new util.HashSet[Integer](zkClient.getSortedBrokerList.map(Integer.valueOf).toSet.asJava)
   }
 
+  override def readProducerId(): util.Optional[java.lang.Long] = {
```

Comment:
This doesn't seem to be returning nextProducerId, it's returning the current producer id.

Why not just have `writeProducerId` take a `ProducerIdsBlock` object so we don't have to do a bunch of dubious translation?

Reply:
Oh right, in KRaft when we give out a new block, we persist the next ID in the log.